### DISCUSSION TITLE
3.9.1 - Two letter ISO country codes dependent on QT ver?

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#2932] [Technical Support & Help] [answered]
> People running servers will benefit from the enhanced handling of --serverinfo: you can now use two-letter ISO country codes (rather than internal Qt numbering).

Is this dependent on what QT version is installed on the server?

I've compiled 3.9.1 on Ubuntu Server 20.04 using `qmake "CONFIG+=noupcasename headless nosound" Jamulus.pro && make clean && make -j4` which built fine. 

The QT version on the server is `5.12.8`.

I then modified my systemd unit to use the ISO country code (CA for Canada) instead of the numeric (38 for Canada) 
```
--serverinfo  "Rick's Open Jam YUL;Montreal 3.9.1;CA"
```
but I get the following on startup:
```
Oct 22 11:11:49 daw6.diix.org jamulus[51279]: Using server info: name = "Rick's Open Jam YUL", city = "Montreal 3.9.1", country/region = "CA" (United States)
```
Reverting back to country code 38 gives the expected result:
```
Oct 22 11:25:18 daw6.diix.org jamulus[51547]: Using server info: name = "Rick's Open Jam YUL", city = "Montreal 3.9.1", country/region = "38" (Canada)
```

Is this a bug or am I misunderstanding something?

**Comment by Rob-NY** [comment, marked as answer]
Try using a lower case "ca" and see if that works.  I have not tried your use-case, but I do see this in a portion of the new code:

`     if ( vstrLocParts.size() >= 2 && vstrLocParts.at ( 1 ).toLower() == sTwoLetterCode )`

**Reply by rdica** [reply]
Thanks Rob, that appears to be it.

**Reply by Rob-NY** [reply]
created issue #2933 and PR #2934
